programs/castle-vault/src/instructions/sync_lp_token_supply.rs

```diff
@@ -0,0 +1,38 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::Mint;
+
+use crate::state::Vault;
+
+#[event]
+pub struct WithdrawEvent {
+    vault: Pubkey,
+    user: Pubkey,
+    amount: u64,
+}
+
+#[derive(Accounts)]
+pub struct SyncLpTokenSupply<'info> {
```

Comment:
Why is this instruction necessary? I think it re-introduces the bug Robert reported before 

https://github.com/castle-finance/castle-vault/pull/51

Reply:
The problem to solve is, `vault.lp_token_supply` is not initialized with the correct value when you update an existing vault that has non-zero allocation. It is initialized to zero instead. We need a way to assign the correct value to `vault.lp_token_supply`.

It would not have been a problem if we create a new vault, because the initial LP token supply is zero anyway.

Reply:
This one requires owner signature, which kind of serves as a safety net. But of course this is not perfect.

Reply:
Gotcha--can you add a comment on this and then it should be good to merge